projects/core/src/cms/page/routing/default-route-page-meta.resolver.ts

```diff
@@ -0,0 +1,67 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import { TranslationService } from '../../../i18n/translation.service';
+import { BreadcrumbMeta } from '../../model/page.model';
+import {
+  RouteBreadcrumbResolver,
+  RouteBreadcrumbResolverParams,
+} from './route-page-meta.model';
+
+/**
+ * Resolves the breadcrumb for the Angular ActivatedRouteSnapshot
+ */
+@Injectable({ providedIn: 'root' })
+export abstract class DefaultRoutePageMetaResolver
+  implements RouteBreadcrumbResolver {
+  constructor(protected translation: TranslationService) {}
+
+  /**
+   * Resolves breadcrumb based on the given url and the breadcrumb config.
+   *
+   * - When breadcrumb config is empty, it returns an empty breadcrumb.
+   * - When breadcrumb config is a string or object with `i18n` property,
+   *    it translates it and use as a label of the returned breadcrumb.
+   * - When breadcrumb config is an object with property `raw`, then
+   *    it's used as a label of the returned breadcrumb.
+   */
+  resolveBreadcrumbs({
```

Comment:
See also my comments below. 

To improve readability, perhaps come up with local method to resolve the label (`getLabel(config: pageMetaConfig)`)? We can then simplify the implementation of `resolveBreadcrumbs()` by just returning:
```typescript
return of([{ link: url, label: getLabel(config) }]);
```